framework/sandstone.cpp

```diff
@@ -3398,26 +3410,34 @@ int main(int argc, char **argv)
     logging_printf(LOG_LEVEL_VERBOSE(1), "THIS IS AN UNOPTIMIZED BUILD: DON'T TRUST TEST TIMING!\n");
 #endif
 
-    generate_test_list();
+    generate_test_list(test_list);
     if (sApp->current_fork_mode() == SandstoneApplication::exec_each_test) {
         disable_test(&mce_test);
     }
 
     // If we want to use the weighted testrunner we need to initialize it
-
     if (test_list_file_path) {
-        test_selector = create_list_file_test_selector(sApp->test_list, test_list_file_path,
-                                                       sApp->starting_test_number, sApp->ending_test_number,
-                                                       sApp->test_list_randomize);
+        test_selector = create_list_file_test_selector(std::move(test_list), test_list_file_path,
+                                                       starting_test_number, ending_test_number,
+                                                       test_list_randomize);
     } else {
         struct weighted_run_info *weighted_test_list = all_weighted_runinfo;
-        if (sApp->test_list_randomize) {
-            test_selector =  create_builtin_test_selector(sApp->test_list, sApp->starting_test_number, sApp->ending_test_number);
-            sApp->use_strict_runtime = true;
+        if (test_list_randomize) {
+            logging_printf(LOG_LEVEL_QUIET, "# WARNING: --test-list-randomize does not do "
+                                            "anything. Ignoring.\n");
+        }
+        if (!test_selector) {
+            switch (test_selection_strategy) {
+            case Alphabetical:
+            case Ordered:
+                test_selector = new OrderedTestSelector(test_list, test_selection_strategy);
```

Comment:
Won't do. I don't want to even compile that file in restricted command-line mode.

Reply:
Not liking the split switch statement, but I get why it is there (to avoid repeating the assertion).  
I'm approving - if I can come up with a better way I'll submit my own update.